src/state/application/updater.ts

```diff
@@ -20,6 +22,40 @@ function useQueryCacheInvalidator() {
   }, [chainId, dispatch])
 }
 
+const NETWORK_HEALTH_CHECK_MS = ms`15s`
+const DEFAULT_MS_BEFORE_WARNING_WAIT = ms`10m`
```

Comment:
nit. not clear that `NETWORK_HEALTH_CHECK_MS` and `DEFAULT_MS_BEFORE_WARNING_WAIT` are both referring to network connectivity.

I think it would be cleaner to create a custom hook `useNetworkConnectivityWarning` that has all this wok and just call it from updater. Similar to `useQueryCacheInvalidator above.